Sources/Sentry/SentryAppStartTracker.m

```diff
@@ -140,19 +131,41 @@ - (void)buildAppStartMeasurement
             return;
         }
 
+        BOOL isPreWarmed = NO;
+        if ([self isActivePrewarmAvailable] && isActivePrewarm) {
+            [SentryLog logWithMessage:@"The app was prewarmed." andLevel:kSentryLevelInfo];
+
+            isPreWarmed = YES;
+        }
+
         // According to a talk at WWDC about optimizing app launch
         // (https://devstreaming-cdn.apple.com/videos/wwdc/2019/423lzf3qsjedrzivc7/423/423_optimizing_app_launch.pdf?dl=1
         // slide 17) no process exists for cold and warm launches. Since iOS 15, though, the system
-        // might decide to pre-warm your app before the user tries to open it. The process start
-        // time returned valid values when testing with real devices if the app start is not
-        // prewarmed. See:
+        // might decide to pre-warm your app before the user tries to open it.
+        // According to the Apple docs "Prewarming executes an app’s launch sequence up until, but
+        // not including, when main() calls UIApplicationMain." Some customer data shows that this
+        // statement is not valid. Prewarming can stop at any of the app launch steps. Our findings
+        // show that most of the prewarmed app starts don't call the main method. Therefore we
+        // subtract the time before the module initialization / main method to calculate the app
+        // start duration. If the app start stopped during a later launch step, we drop it below
+        // with checking the SENTRY_APP_START_MAX_DURATION. With this approach, we will lose some
+        // warm app starts, but we accept this tradeoff. Useful resources:
```

Comment:
m: I think that 

> Prewarming executes an app’s launch sequence up until, but not including, when main() calls UIApplicationMain 

is consistent with 

> Our findings show that most of the prewarmed app starts don't call the main method

I'd just state our summary of the situation as we've observed it:

```suggestion
        // Prewarming can stop at any of the app launch steps. Our findings show that most of
        // the prewarmed app starts don't call the main method. Therefore we subtract the 
        // time before the module initialization / main method to calculate the app start 
        // duration. If the app start stopped during a later launch step, we drop it below with 
        // checking the SENTRY_APP_START_MAX_DURATION. With this approach, we will 
        // lose some warm app starts, but we accept this tradeoff. Useful resources:
```